gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/TokenResource.java

```diff
@@ -527,6 +527,12 @@ public Response revoke(String token) {
     return resp;
   }
 
+  private boolean triesToRevokeOwnToken(String tokenId, String revoker) throws UnknownTokenException {
+    final TokenMetadata metadata = tokenStateService.getTokenMetadata(tokenId);
+    final String tokenUserName = metadata == null ? "" : metadata.getUserName();
+    return revoker != null && revoker.equals(tokenUserName);
```

Comment:
Revoker is never an empty string, right?

Reply:
In theory, it should always be set, but you are right: an additional emptiness checking does not hurt and could avoid weird issues.
I submitted a new PS.